icicle-compiler/test/Icicle/Test/Sea/Psv.hs

```diff
@@ -174,16 +179,39 @@ failWithError' prints wt = \case
 ------------------------------------------------------------------------
 
 data ShowInput = ShowInputOnError | ShowInputOnSuccess
-  deriving (Eq)
+  deriving (Eq, Show)
 
 data ShowOutput = ShowOutputOnError | ShowOutputOnSuccess
-  deriving (Eq)
+  deriving (Eq, Show)
 
+data PsvOpts  = PsvOpts Int Int Int
+  deriving (Show)
 
 data TestOpts = TestOpts ShowInput ShowOutput S.PsvInputFormat S.InputAllowDupTime
-
-runTest :: WellTyped -> TestOpts -> EitherT S.SeaError IO ()
-runTest wt (TestOpts showInput showOutput inputFormat allowDupTime) = do
+  deriving (Show)
+
+data WellTypedPsv = WellTypedPsv WellTyped PsvOpts
+  deriving (Show)
+
+instance Arbitrary WellTypedPsv where
+  arbitrary = do
+    wt <- arbitrary
+    psv <- genPsvOpts wt
+    return $ WellTypedPsv wt psv
+
+genPsvOpts :: WellTyped -> Gen PsvOpts
+genPsvOpts wt = do
+  -- maximum number of rows to read before compute
+  let row x = x + 1
+  maxRowCount <- row . getPositive <$> arbitrary
+  -- the buffer needs to be at least as large as a single line
+  let str x = x + longestLine wt + 4
```

Comment:
I'm not sure what the sensible behaviour in that case is. Right now I believe ice/icicle-bench segfaults.